packages/amazon-pay-integration/src/amazon-pay-v2-button-options.ts

```diff
@@ -0,0 +1,31 @@
+import {
+    AmazonPayV2ButtonConfig,
+    AmazonPayV2ButtonParameters,
+} from '@bigcommerce/checkout-sdk/amazon-pay-utils';
+import { BuyNowCartRequestBody } from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+export function isWithBuyNowFeatures(options: unknown): options is WithBuyNowFeature {
+    if (!(options instanceof Object)) {
+        return false;
+    }
+
+    return 'buyNowInitializeOptions' in options;
+}
```

Comment:
It might be better to move this typeguard to a separate file 👀

Reply:
done